CMakeLists.txt

```diff
@@ -50,6 +50,16 @@ set(CONFORMANCE_SUFFIX "" )
 include(CMakeVendor.txt OPTIONAL)
 
 # CL_INCLUDE_DIR - path to dir with OpenCL headers
+find_package(OpenCL REQUIRED)
+if(NOT (CL_INCLUDE_DIR AND CL_LIB_DIR) AND OpenCL_FOUND)
+    set(CL_INCLUDE_DIR OpenCL_INCLUDE_DIRS)
+    set(CL_LIB_DIR OpenCL_LIBRARY)
+    message(STATUS "Using system OpenCL: Include DIRS = ${CL_INCLUDE_DIR}")
+else()
+    message(STATUS "Using user defined OpenCL: Include Dirs = ${CL_INCLUDE_DIR}")
+    message(STATUS "Using user defined OpenCL: lib Dir = ${CL_LIB_DIR}")
+endif()
```

Comment:
This CMake change looks unrelated to the `clEnqueueFillBuffer` changes, I think it should be a standalone PR if this is a change to the build system you'd like to see.

Reply:
Agree that build changes should be separated out into their own PR

Sreelakshmi Haridas Maruthur
Qualcomm

Reply:
I've removed it for now.